pkg/expression/builtin_ilike_vec.go

```diff
@@ -52,28 +53,30 @@ func (b *builtinIlikeSig) canMemorize(param *funcParam) bool {
 	return param.getCol() == nil
 }
 
-func (b *builtinIlikeSig) tryToMemorize(param *funcParam, escape int64) {
+func (b *builtinIlikeSig) tryToVecMemorize(ctx EvalContext, param *funcParam, escape int64) (collate.WildcardPattern, bool) {
 	if !b.canMemorize(param) {
-		return
+		return nil, false
 	}
 
-	memorization := func() {
-		if b.pattern == nil {
-			b.pattern = collate.ConvertAndGetBinCollation(b.collation).Pattern()
-			b.pattern.Compile(param.getStringVal(0), byte(escape))
-			b.isMemorizedPattern = true
-		}
+	pattern, err := b.patternCache.getOrInitCache(ctx, func() (collate.WildcardPattern, error) {
+		pattern := collate.ConvertAndGetBinCollation(b.collation).Pattern()
+		pattern.Compile(param.getStringVal(0), byte(escape))
+		return pattern, nil
+	})
+
+	intest.AssertNoError(err)
+	if err != nil {
+		return nil, false
 	}
 
-	// Only be executed once to achieve thread-safe
-	b.once.Do(memorization)
+	return pattern, true
 }
 
 func (b *builtinIlikeSig) getEscape(ctx EvalContext, input *chunk.Chunk, result *chunk.Column) (int64, bool, error) {
 	rowNum := input.NumRows()
 	escape := int64('\\')
 
-	if !b.args[2].ConstItem(ctx.GetSessionVars().StmtCtx.UseCache) {
+	if !b.args[2].ConstItem(true) {
 		return escape, true, errors.Errorf("escape should be const")
```

Comment:
This line cannot never reached because the parser blocked all other formats except for a literal format. I just use `ConstItem(true)` to make it a some behavior with parser